src/NServiceBus.Core.Tests/Msmq/MsmqQueueCreatorTests.cs

```diff
@@ -247,12 +229,16 @@ void DeleteQueueIfPresent(string queueName)
         {
             var path = MsmqAddress.Parse(queueName).PathWithoutPrefix;
 
-            if (!MessageQueue.Exists(path))
+            if (MessageQueue.Exists(path))
             {
-                return;
+                MessageQueue.Delete(path);
             }
+        }
 
-            MessageQueue.Delete(path);
+        string GetReallyLongQueueName()
+        {
+            var maxQueueNameForSetPermissionToWork = 102 - Environment.MachineName.Length;
```

Comment:
@andreasohlund - As long as the entire format name (machine name\private$\[queueName]) is 124 or less, setting queue permissions should work. In this case, the test is making sure that it will always set a valid length, so the test will always pass? 
The current implementation does not throw an exception based on queue length, so I am not sure how calling SetQueuePermissions will cause this backward compatibility issue. I understand your point and that is very relevant for the new issue I raised, but for this PR, not sure if the test is relevant. Perhaps I am missing something. Let's get on a call.